apps/HelloHexagon/Makefile

```diff
@@ -43,13 +43,18 @@ $(BIN)/%/pipeline_hvx128.o: $(BIN)/pipeline
 $(BIN)/process-%: process.cpp $(BIN)/%/pipeline_cpu.o $(BIN)/%/pipeline_hvx64.o $(BIN)/%/pipeline_hvx128.o
 	$(CXX-$*) $(CXXFLAGS) $(CXXFLAGS-$*) -I $(BIN)/$* -Wall -O3 process.cpp $(BIN)/$*/pipeline_cpu.o $(BIN)/$*/pipeline_hvx64.o $(BIN)/$*/pipeline_hvx128.o -o $(BIN)/process-$* $(LDFLAGS-$*)
 
-# This requires root: adb root && adb wait-for-device && adb remount
+DEVICE_PATH ?= /data/local/tmp/HelloHexagon
+DEVICE_ENV = "LD_LIBRARY_PATH=$(DEVICE_PATH):/vendor/lib64 ADSP_LIBRARY_PATH=\"$(DEVICE_PATH);/dsp\""
+HEXAGON_RUNTIME_PATH = $(HALIDE_SRC_PATH)/src/runtime/hexagon_remote
 run-%-android: $(BIN)/process-%-android
-	adb push $(BIN)/process-$*-android /data/
-	adb shell chmod +x /data/process-$*-android
-	adb shell /data/process-$*-android cpu 10
-	adb shell /data/process-$*-android hvx64 10
-	adb shell /data/process-$*-android hvx128 10
+	adb shell mkdir -p $(DEVICE_PATH)
```

Comment:
It would be nice to package up some of this as a reusable script or rule (in a subsequent PR) since there's a lot of overlap between the app examples here.